backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -989,4 +990,107 @@ async def check_duplicate_tickets(
+        incidents_sheet_name: str = Form(default="Incidents",
+                                         description="Name of the sheet containing incident data"),
+        mapping_type_subtype_file: UploadFile = File(...),
+        mapping_type_subtype_sheet_name: str = Form(default="Mapping Old_New_v1.0"),
+        request_info: str = Form(default="", description="Request info in JSON format")
+):
+    temp_file = None
+    request_info = request_info_from_json(request_info)
+    get_authorized_request_info(request_info)
+
+    try:
+        temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=".xlsx")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)